packages/flutter/test/material/text_selection_test.dart

```diff
@@ -61,5 +61,15 @@ void main() {
       ));
       expect(materialTextSelectionControls.canSelectAll(key.currentState), false);
     });
+
+    testWidgets('should return true if can copy contents', (WidgetTester tester) async {
+      final GlobalKey<EditableTextState> key = GlobalKey();
+      await tester.pumpWidget(createEditableText(
+        key: key,
+        text: '123 456',
+        selection: const TextSelection(baseOffset: 0, extentOffset: 3),
+      ));
+      expect(materialTextSelectionControls.canCopy(key.currentState), true);
+    });
```

Comment:
This PR will need more tests when it's ready, maybe some like this:

  1. Set a thin screen width and see that a selection menu button is missing. Verify that you can open the submenu and still tap "Select All".
  1. Verify that the submenu's back button works.
  1. Verify that very narrow screen widths hide more than 1 button in the submenu.
  1. Verify that a submenu near the top/bottom of the screen doesn't extend beyond the edges.
  1. Maybe show that changing the locale can change the number of buttons in the selection menu.

I can help point you in the right direction for how to write these when we get there!

Reply:
Well, should the submenu be pop up menu or what I have done is fine??

Reply:
I think the widget structure that you have for the submenu looks good, and conditionally rendering either `_buildSelectionToolbar` or `_buildSelectionMenu` also looks good.